### ISSUE TITLE
Trying to install non-native Linux "demo" result in a "Invalid platform", while free/paid games don't have such issue

### ISSUE THREAD

**Issue by braiam** [ValveSoftware/steam-for-linux#10899]
I would expect that the store allowed me to add to my library any game I find on the store. This works fine with paid games (add to cart) and free games (add to library), but demos are _different_. It seems that they check the platform, which would only be avoided if I enable to use compatibility tools for all other titles. The web/browser store doesn't have these kinds of issues, only the Steam client.

TL;dr: Steam client shouldn't check the platform when adding a demo to the library.

**Comment by braiam** [comment]
Sorry @kisak-valve but I disagree. I read that one and is just a amalgamation of issues that should be instead tracked as concrete and deliberate issues, starting with the issue that that option shouldn't effect what I add to my library on the steam client from the store page. I can't claim I know how it's codified behind the scenes, but compatibility layers should only be considered on applications in my library, not before.

I'm not asking for "clarification" about what the "other" toggle does, I'm asking that everything that has to do with adding applications to my library would be unencumbered by platform checks.

Also, I made a mistake. The web store behaves the same way, the issue was that I used Euro Truck simulator that wasn't clear to me that it had a native client. So, users that want to add demos only have steamcmd as option to add them to their library. Enabling and disabling compatibility layers just to test out demos break the user flow, since the client needs a restart to toggle it, which would defect the purpose of demos.

**Comment by braiam** [comment]
@kisak-valve considering the release notes of the new revamped demo experience, I should consider this issue solved, rather than related to that other issue:

> You can add demos to your library without immediately installing them. You'll find an 'add to library' button appearing for demos now that let you add to your library from the mobile app or other places where you may not be ready to install them immediately.

From https://steamcommunity.com/groups/steamworks/announcements/detail/4155211502162971563

That other issue may remain as it is.

**Comment by rept0id** [comment]
1. "Steam" > "Settings" > "Steam Play" > "Enable Steam Play for All Titles"
2. Restart steam - if this wont happen automatically
3. Navigate to the demo. Now you'll be able to download it

**Comment by boyreau** [comment]
"Steam Play" was renamed to "Compatibility"
